common/git-hooks/pre-commit

```diff
@@ -0,0 +1,19 @@
+#!/bin/sh
+# Called by "git commit" with no arguments.  The hook should
+# exit with non-zero status after issuing an appropriate message if
+# it wants to stop the commit.
+
+COMMAND=common/temp/node_modules/.bin/precise-commits
+
+echo --------------------------------------------
+echo Starting Git hook: pre-commit
+
+if [ -f $COMMAND ]; then
+  echo Invoking $COMMAND
+  $COMMAND
```

Comment:
I think we should consider capturing the exit status here and passing it through at the end.

Reply:
This could cause the commit to fail, which conflicts with assumption 4 in my [comment](https://github.com/Azure/azure-sdk-for-js/pull/10219#issuecomment-663737624) below. I am happy to change this if we get a consensus on another approach with another set of assumptions.